aquadoggo/src/libp2p/identity.rs

```diff
@@ -0,0 +1,111 @@
+// SPDX-License-Identifier: AGPL-3.0-or-later
+
+use std::fs;
+use std::fs::File;
+use std::io::prelude::*;
+use std::os::unix::fs::PermissionsExt;
+use std::path::Path;
+
+use anyhow::Result;
+use libp2p::identity::{ed25519, Keypair};
+use libp2p::PeerId;
+use pem::Pem;
+
+/// Utilities for dealing with a libp2p identity in the form of an ed25519 keypair.
+pub trait Identity {
+    fn new() -> Self
+    where
+        Self: Sized;
+    fn id(&self) -> PeerId;
+    fn keypair(&self) -> Keypair;
+    fn encode_pem(&self) -> String;
+    fn save(&self, path: &Path) -> Result<()>;
+    fn load(path: &Path) -> Result<Self>
+    where
+        Self: Sized;
+}
+
+impl Identity for Keypair {
+    /// Generate a random keypair.
+    fn new() -> Self {
+        Keypair::generate_ed25519()
+    }
+
+    /// Return the public key of a keypair.
+    fn id(&self) -> PeerId {
+        PeerId::from(self.public())
+    }
+
+    /// Return the keypair.
+    fn keypair(&self) -> Keypair {
+        self.clone()
+    }
+
+    /// Encode the keypair using the PEM (Privacy-Enhanced Mail) format.
+    fn encode_pem(&self) -> String {
```

Comment:
I chose to use an existing key-storage format rather than chucking the keys into a `TOML` file (or similar).

Reply:
Is there a way to store only the private key as an hex-encoded string? It would allow for easier access, especially if you might want to copy it over in another context (thats not really what we want users to do, but even just for debugging its helpful).

I'd prefer that and then convert it all towards libp2p in the way they want, but towards us a simple text file could be nice (similar to how `send-to-node` does it).

Optionally we can also use "our" identity API to generate the key pair: https://docs.rs/p2panda-rs/latest/p2panda_rs/identity/index.html - to have everything coming out of one hand?